server/manager/src/main/java/org/apache/accumulo/manager/compaction/coordinator/CompactionCoordinator.java

```diff
@@ -1070,21 +1071,30 @@ private void cleanUpCompactors() {
       var groups = zoorw.getChildren(compactorQueuesPath);
 
       for (String group : groups) {
-        String qpath = compactorQueuesPath + "/" + group;
-
-        var compactors = zoorw.getChildren(qpath);
+        final String qpath = compactorQueuesPath + "/" + group;
+        final CompactorGroupId cgid = CompactorGroupId.of(group);
+        final var compactors = zoorw.getChildren(qpath);
 
         if (compactors.isEmpty()) {
           deleteEmpty(zoorw, qpath);
-        }
-
-        for (String compactor : compactors) {
-          String cpath = compactorQueuesPath + "/" + group + "/" + compactor;
-          var lockNodes = zoorw.getChildren(compactorQueuesPath + "/" + group + "/" + compactor);
-          if (lockNodes.isEmpty()) {
-            deleteEmpty(zoorw, cpath);
+          // Group has no compactors, we can clear its
+          // associated priority queue of jobs
+          getJobQueues().getQueue(cgid).clear();
+        } else {
+          int aliveCompactorsForGroup = 0;
+          for (String compactor : compactors) {
+            String cpath = compactorQueuesPath + "/" + group + "/" + compactor;
+            var lockNodes = zoorw.getChildren(compactorQueuesPath + "/" + group + "/" + compactor);
+            if (lockNodes.isEmpty()) {
+              deleteEmpty(zoorw, cpath);
+            } else {
+              aliveCompactorsForGroup++;
+            }
           }
+          getJobQueues().getQueue(cgid).setMaxSize(
+              Math.min((int) (aliveCompactorsForGroup * queueSizeFactor), Integer.MAX_VALUE));
```

Comment:
Its possible that a queue may not exist if there are no jobs.  Could add the following method to CompactionJobQueues.  Suggested this because there is a getQueueMaxSize() method.

```java
  public void setQueueMaxSize(CompactorGroupId groupId, int size) {
    var prioQ = priorityQueues.get(groupId);
    if(!prioQ != null){ prioQ.setMaxSize(size)};
  }
```

Then call it here.  

```suggestion
          getJobQueues().setQueueMaxSize(cgid, Math.min((int) (aliveCompactorsForGroup * queueSizeFactor), Integer.MAX_VALUE));
```

Reply:
Handled NPE in 18d2243